airflow/api_fastapi/parameters.py

```diff
@@ -129,6 +129,26 @@ def transform_aliases(self, value: str | None) -> str | None:
         return value
 
 
+class _OrderByParam(BaseParam[str]):
```

Comment:
Since we only need the name field in DagTag, I think passing asc or desc as a query parameter for ordering is more straightforward. This could also be useful for similar use cases where only a single field needs to be ordered in a model.

Reply:
I don't think we need that, this is really niche because most of the list endpoints have multiple order possibility.

Then as a user you have to think:
- Is there multiple sort possiblities, then use `?order_by=xxx` put a field with `-` and ` `
- There is not, then user `?order_by=xxx, with `asc` and `desc`

Mixing two concept in the same query_param.

I think we should remove it and use the systematic `order_by=field` or `order_by=-field`

Reply:
Hi @pierrejeambrun, I encountered an issue with `SortParm` when working on PostgreSQL. The error details are as follows:

```
2024-10-17 16:30:52.868 UTC [15780] ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list at character 84
2024-10-17 16:30:52.868 UTC [15780] STATEMENT:  SELECT DISTINCT dag_tag.name 
        FROM dag_tag 
        WHERE dag_tag.name ILIKE '%%' ORDER BY CASE WHEN (dag_tag.name IS NOT NULL) THEN 0 ELSE 1 END, dag_tag.name ASC, dag_tag.name 
         LIMIT 2 OFFSET 1
```

I haven't yet thought of a solution that avoids distinguishing the dialect type for returning a specific SQL statement.

Reply:
I cannot reproduce the error on your branch. Both unit tests (with sqllite), and manual local tests with breeze on postgres work:
![Screenshot 2024-10-18 at 14 13 53](https://github.com/user-attachments/assets/a682b257-35e7-45ab-b9a9-49158e79bcd8)

What are the steps to experience your issue ?

Reply:
Yes, I think it's because of the `distinct` requirering the `case` statement in the select. `group_by` happens before the projection and seems to solve the problem.

`select(DagTag.name).group_by(DagTag.name)` instead of `select(distinct(DagTag.name))`.